hamt/hamt.go

```diff
@@ -81,13 +77,16 @@ type Shard struct {
 
 // NewShard creates a new, empty HAMT shard with the given size.
 func NewShard(dserv ipld.DAGService, size int) (*Shard, error) {
+	return NewShardWithHashFunc(dserv, size, mh.MURMUR3_128)
+}
+
+func NewShardWithHashFunc(dserv ipld.DAGService, size int, hashFunc uint64) (*Shard, error) {
```

Comment:
I agree that this is better than a global variable. However, I think this function should not be exported as it will let people think this library actually supports multiple hash functions which it does not currently do (and hasn't been specced for yet).

If this function can be unexported and just used by tests than 🎉. However, if our options are:
1. Move all of hamt.go to the internal package
2. Export a test function people could easily misuse to create invalid data
3. Use a global in the internal package that will only be used in a couple of tests

then option 3 seems ok to me. If you have other suggestions/options I'm open to them though.